css/css-display/accessibility/display-contents-role-and-label.html

```diff
@@ -0,0 +1,290 @@
+  <div role="button" tabindex="0" style="display: contents;" data-expectedrole="button" data-expectedlabel="x" data-testname="div with button role, tabindex=0 and display: contents has button role" class="ex-role-and-label">x</div>
+
+  <!-- Display: Flex -->
+  <table style="display: flex; flex-flow: column nowrap;" data-expectedrole="table">
+      <thead>
+        <tr style="display: contents;" data-expectedrole="row" data-testname="tr with display: contents, within table with display: flex, has row role" class="ex-role">
+          <th data-expectedrole="columnheader" data-expectedlabel="x" data-testname="th as child of tr with display: contents, within table with display: flex, has columnheader role" class="ex-role-and-label">x</th>
+          <th>x</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr style="display: contents;">
+          <td data-expectedrole="cell" data-expectedlabel="x" data-testname="td as child of tr with display: contents, within table with display: flex, has cell role" class="ex-role-and-label">x</td>
+          <td>x</td>
+        </tr>
+      </tbody>
+  </table>
+
+  <table role="table" style="display: flex; flex-flow: column nowrap;" data-expectedrole="table">
+    <thead>
+      <tr style="display: contents;" data-expectedrole="row" data-testname="tr with display: contents, within table with role=table with display: flex, has row role" class="ex-role">
+        <th data-expectedrole="columnheader" data-expectedlabel="x" data-testname="th as child of tr with display: contents, within table with role=table with display: flex, has columnheader role" class="ex-role-and-label">x</th>
+        <th>x</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr style="display: contents;">
+        <td data-expectedrole="cell" data-expectedlabel="x" data-testname="td as child of tr with display: contents, within table with role=table with display: flex, has cell role" class="ex-role-and-label">x</td>
+        <td>x</td>
+      </tr>
+    </tbody>
+  </table>
+
+  <table role="grid" style="display: flex; flex-flow: column nowrap;" data-expectedrole="grid">
+    <thead>
+      <tr style="display: contents;" data-expectedrole="row" data-testname="tr with display: contents, within table with role=grid with display: flex, has row role" class="ex-role">
+        <th data-expectedrole="columnheader" data-expectedlabel="x" data-testname="th as child of tr with display: contents, within table with role=grid with display: flex, has columnheader role" class="ex-role-and-label">x</th>
+        <th>x</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr style="display: contents;">
+        <td data-expectedrole="gridcell" data-expectedlabel="x" data-testname="td as child of tr with display: contents, within table with role=grid with display: flex, has gridcell role" class="ex-role-and-label">x</td>
```

Comment:
I need to look this one up. You may have found another instance of ambiguity in the spec wrt contextual roles. Since table is NOT a grid, adding grid to the table may not require inferring the cascade of contextual sub-elements' roles.

```suggestion
        <td data-expectedrole="gridcell" data-expectedlabel="x" data-testname="td (no explicit role) as child of tr with display: contents, within table with role=grid with display: flex, has gridcell role" class="ex-role-and-label">x</td>
```